cmd/grype/cli/options/datasources.go

```diff
@@ -6,20 +6,24 @@ import (
 
 const (
 	defaultMavenBaseURL = "https://search.maven.org/solrsearch/select"
+	defaultAbortAfter   = "10m"
 )
 
 type externalSources struct {
-	Enable bool  `yaml:"enable" json:"enable" mapstructure:"enable"`
-	Maven  maven `yaml:"maven" json:"maven" mapstructure:"maven"`
+	Enable     bool   `yaml:"enable" json:"enable" mapstructure:"enable"`
+	AbortAfter string `yaml:"abort-after" json:"abortAfter" mapstructure:"abort-after"`
+	Maven      maven  `yaml:"maven" json:"maven" mapstructure:"maven"`
 }
 
 type maven struct {
-	SearchUpstreamBySha1 bool   `yaml:"search-upstream" json:"searchUpstreamBySha1" mapstructure:"search-maven-upstream"`
-	BaseURL              string `yaml:"base-url" json:"baseUrl" mapstructure:"base-url"`
+	SearchUpstreamBySha1 bool    `yaml:"search-upstream" json:"searchUpstreamBySha1" mapstructure:"search-maven-upstream"`
+	BaseURL              string  `yaml:"base-url" json:"baseUrl" mapstructure:"base-url"`
+	AbortAfter           *string `yaml:"abort-after" json:"abortAfter" mapstructure:"abort-after"`
 }
 
 func defaultExternalSources() externalSources {
 	return externalSources{
+		AbortAfter: defaultAbortAfter,
 		Maven: maven{
 			SearchUpstreamBySha1: true,
 			BaseURL:              defaultMavenBaseURL,
```

Comment:
We want the following quality to multi-teir config items like this:

-  a user specifying top-level fields overrides UNspecified nested fields
-  a user specifying top-level fields does NOT override specified nested fields
-  a user specifying nested fields does NOT top-level fields

Where:
- specified: a user explicitly provided a value
- unspecified: a user did NOT explicitly provide a value. This might mean no value should be set or a default value should be set instead.

The implementation does follow this, however, the rendered configuration unintentionally conveys that there is no maven abort-after value:
```yaml
# running: grype -vv
...
  external-sources:
      abort-after: 10m
      maven:
          abort-after: null
```

Here from a user perspective it appears as if an override has been set at the top level, but has not been inherited. This is because we're conflating inheriting default values vs overriding nested values.

What we really want is to convey that the user did not specify overriding values in top-level fields, and that the individual sources have sane defaults:

```yaml
# running: grype -vv
...
  external-sources:
      abort-after: null
      maven:
          abort-after: 10m0s
```

This has the added benefit that different sources can have different default values (since the default value for nested properties is not inheriting a top-level property).

We've dealt with multi-level properties in syft, see the `pretty` fields for an example: https://github.com/anchore/syft/blob/v1.4.1/cmd/syft/internal/options/format.go . We end up making both fields optional to show what is specified vs unspecified, then reason about the final values in a `PostLoad` function, which is invoked while rendering the configuration.

Specifying an override then appears in both places correctly:
```yaml
# run: GRYPE_EXTERNAL_SOURCES_ABORT_AFTER=40m grype -vv 
  external-sources:
      abort-after: 40m0s
      maven:
          abort-after: 40m0s
```

Instead of no value being set:
```yaml
# run: GRYPE_EXTERNAL_SOURCES_ABORT_AFTER=40m grype -vv 
  external-sources:
      abort-after: 40m
      maven:
          abort-after: null
```

Reply:
FYI, no action needed, I made the adjustment to use the `PostLoad` in a follow up commit.